python/tk_multi_launchapp/launch_apps.py

```diff
@@ -0,0 +1,651 @@
+        paths may not contain a task, so this may need to be pushed through
+        separately via the context.
+        """
+        if context is None:
+            # this context looks sour. So fall back on to path-only launch.
+            self.launch_from_path(path, version)
+        else:
+            # use given context to launch engine!
+            self._launch_app(context, path, version)
+
+
+class SoftwareEntityLauncher(BaseLauncher):
+    """
+    Launches a DCC based on site Software entity entries.
+    """
+    def init_launch_commands(self):
+        """
+        Determine what launch command(s) to register with the current TK engine.
+        Multiple commands may be registered based on the number of retrieved
+        Software entities and their corresponding 'versions' field.
+        """
+        # Expand Software field names that rely on the current platform
+        app_path_field = "sg_%s_path" % self._platform_name
+        app_args_field = "sg_%s_args" % self._platform_name
+
+        # Determine the information to retrieve from SG
+        sw_entity = self._tk_app.get_setting("sg_software_entity") or "Software"
+        sw_filters = [
+            ["sg_status_list", "is", "act"],
+        ]
+        sw_fields = [
+            app_path_field,
+            app_args_field,
+            "code",
+            "image",
+            "sg_engine",
+            "sg_group_restrictions",
+            "sg_projects",
+            "sg_versions",
+            "sg_user_restrictions",
+        ]
+
+        # Get the list of Software apps that match the specified filters.
+        sw_entities = self._tk_app.shotgun.find(
+            sw_entity, sw_filters, sw_fields
+        )
+        if not sw_entities:
+            # No Entities found matching filters, nothing to do.
+            self._tk_app.log_debug("No SG %s entities found matching filters : %s" %
+                (sw_entity, sw_filters)
+            )
+            return
+
+        # Short-hand for a couple of useful context values
+        ctx_user = self._tk_app.context.user
+        ctx_project = self._tk_app.context.project
+
+        for sw_app in sw_entities:
+            app_menu_name = sw_app["code"]
+            app_engine = sw_app["sg_engine"]
+            app_path = sw_app[app_path_field]
+            app_args = sw_app[app_args_field]
+            app_projects = sw_app["sg_projects"] or []
+            app_users = sw_app["sg_user_restrictions"] or []
+            app_groups = sw_app["sg_group_restrictions"] or []
+            app_icon = sw_app["image"]
+
+            if not app_path:
+                # If no path has been set for the app, we will eventually go look for one,
+                # but for now, don't load the app.
+                self._tk_app.log_warning("No path specified for app [%s]." % app_menu_name)
+                continue
+
+            if app_engine and app_engine not in self._tk_app.engine.get_env().get_engines():
```

Comment:
`self._tk_app.engine.get_env().get_engines()` is accesssing internal core methods that are not part of the [public API](http://developer.shotgunsoftware.com/tk-core/platform.html#engines). Could we use [get_engine_path](http://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.get_engine_path) instead? Or do we need something new to core?